docs/semantics.md

```diff
@@ -248,6 +248,16 @@ However, implicit directories does have drawbacks:
 
 Alternatively, users can create a script which lists the buckets and creates the appropriate objects for the directories so that the ```--implicit-dirs``` flag is not used.
 
+**Using Cloud Storage Buckets with Hierarchical Namespaces Enabled :**
+
+Cloud Storage Fuse also offers seamless support for buckets with hierarchical namespaces enabled. Mounting an HNS-enabled bucket using gcsfuse works exactly like mounting a standard bucket, with no additional configuration required.
+
+HNS-enabled buckets offer several advantages over standard buckets when used with cloud storage fuse:
+
+- HNS-enabled buckets eliminates the need for --implicit-dirs flag. HNS buckets inherently understand directories, so gcsfuse no longer needs this flag to simulate directory structures. Users will see consistent directory listings with or without the flag.
```

Comment:
we could add gcsfuse does not need to simulate directories using placeholder objects ( 0-byte objects ending with '/' )

Reply:
done